capstone/src/components/User/User.jsx

```diff
@@ -0,0 +1,34 @@
+import './User.css';
+import { useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import UserContext from '../../Contexts/UserContext';
+import Header from './Header/Header';
+
+export function userPage(userData) {
+  const navigate = useNavigate();
+  if (Object.keys(userData).length === 0) {
+    return (<h2>Loading ...</h2>);
+  }
+  if (userData.data.userType === 'adventurer') {
+    navigate('/adventurer');
+    return (<h2>You logged in as an Adventurer</h2>);
+  }
+  if (userData.data.userType === 'experienceMaker') {
+    navigate('/experience');
+    return (<h2>You logged in as an Experience Maker</h2>);
+  }
+
+  return (<h2>An error has occurred</h2>);
+}
+
+export default function User({ setIsLoggedIn }) {
+  const userData = useContext(UserContext);
+  return (
+    <div className="user">
+      <Header setIsLoggedIn={setIsLoggedIn} />
+      {userPage(userData)};
```

Comment:
It's subtle, but let's avoid calling functions that return JSX components. Instead we can just make `userPage` itself a component.

```
<Header ... />
<UserPage />
```

Second, once userPage is created as a JSX component, you can then use hooks like  `useContext(UserContext)` *directly inside UserPage*, so you no longer need props/arguments.